.github/workflows/kit-new.yml

```diff
@@ -0,0 +1,165 @@
+                kubectl logs deploy/kit-controller -n kit 
+                exit 1
+              fi
+              sleep 5
+            done
+            kubectl --kubeconfig=$GUESTKUBECONFIG apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/release-1.10/config/master/aws-k8s-cni.yaml
+            kubectl --kubeconfig=$GUESTKUBECONFIG rollout status daemonset/aws-node -n kube-system
+            for (( time=0; time<=160; time+=5 )); do
+              ready_node=$(kubectl --kubeconfig=$GUESTKUBECONFIG get nodes 2>/dev/null | grep -w Ready | wc -l)
+              if [[ "$ready_node" -eq 2 ]]; then
+                echo "Dataplane nodes are ready"
+                break;
+              else
+                echo "Dataplane nodes are not ready"
+              fi
+              if [[ "$time" -eq 150 ]]; then
+                echo "Timeout: Dataplane nodes are not ready after 3 minutes"
+                exit 1
+              fi
+              sleep 5
+            done
+            kubectl --kubeconfig=$GUESTKUBECONFIG create deployment nginx-deployment --image=nginx
+            kubectl --kubeconfig=$GUESTKUBECONFIG rollout status deployment/nginx-deployment
+            kubectl --kubeconfig=$GUESTKUBECONFIG expose deployment nginx-deployment --name=nginx-service --port=80 --target-port=80 --type=ClusterIP
+            kubectl --kubeconfig=$GUESTKUBECONFIG certificate approve $(kubectl --kubeconfig=$GUESTKUBECONFIG get csr | grep Pending | awk '{print $1}')
+            kubectl --kubeconfig=$GUESTKUBECONFIG run -i --rm --restart=Never curly-pod  --image=curlimages/curl --command -- curl -s 'http://nginx-service:80'
+          done
+      - name:  Delete the kit environment 
+        if: always()
+        run: | 
+              #During the first attempt, "kitctl" was unable to delete some resources sporadically. To ensure that enough time is given for the resources to be deleted, a longer wait time of 220 seconds is added before attempting to delete the kit again.
+              #and for the remaining 3 iterations with a wait time of 60 seconds to ensure all resources are successfully deleted as we are left with few resources sometimes.
+              allVersions=("1-21")
+              cd ${HOME}/work/kubernetes-iteration-toolkit/kubernetes-iteration-toolkit/substrate/cmd/kitctl
+               if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+                os_type="linux"    
+              elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+                os_type="darwin"
+              fi 
+              export KUBECONFIG=${HOME}/.kit/env/kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}/etc/kubernetes/admin.conf
+              kubectl config current-context
+              for version in "${allVersions[@]}"; do
+                GUEST_CLUSTER_NAME="guest-${os_type}-${versions}-${{ github.run_id }}-${{github.run_attempt}}"
+                kubectl delete controlplane ${GUEST_CLUSTER_NAME}
```

Comment:
I'd suggest splitting CP cleanup from kit env cleanup, since we want to ensure we execute both cleanup no matter if one of them fails.
Currently if `kubectl delete controlplane`, the cleanup for kit will not be executed.